lib/imagec/download.go

```diff
@@ -256,8 +257,13 @@ func (ldm *LayerDownloader) makeDownloadFunc(layer *ImageWithMeta, ic *ImageC, p
 				}
 			}
 
+			// Use the new context (created as part of Transfer) and the operation ID
+			// passed in from the callers
+			opID := op.ID()
+			op := trace.NewOperationFromID(d.Transfer.Context(), &opID, "makeDownloadFunc")
```

Comment:
I'm not sure this shadowing of the `op` from the outer-outer function is a good idea. Can we make this new operation just after we initialize the new `downloadTransfer` structure and/or call it something different depending on whether it's supposed to override the outermost `op`.